lib/appsignal/heartbeat.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require "net/http"
+require "json"
+
+module Appsignal
+  class Heartbeat
+    class << self
+      def transmitter
+        @transmitter ||= Appsignal::Transmitter.new(
+          "#{Appsignal.config[:logging_endpoint]}/heartbeats/json"
+        )
+      end
+    end
+
+    attr_reader :name, :id
+
+    def initialize(name:)
+      @name = name
+      @id = SecureRandom.hex(8)
+    end
+
+    def start
+      transmit_event("start")
+    end
+
+    def finish
+      transmit_event("finish")
+    end
+
+    private
+
+    def event(kind)
+      {
+        :name => name,
+        :id => @id,
+        :kind => kind,
+        :timestamp => Time.now.utc.to_i
+      }
+    end
+
+    def transmit_event(kind)
+      unless Appsignal.active?
+        Appsignal.internal_logger.debug("AppSignal not active, not transmitting heartbeat event")
```

Comment:
This needs and early return if you do not want it to send the event. Can you add a test for this scenario too?

```suggestion
        Appsignal.internal_logger.debug("AppSignal not active, not transmitting heartbeat event")
        return
```

Reply:
Big whoops here, thanks for noticing it. Will do.